### ISSUE TITLE
Roberta output changing between calls

### ISSUE THREAD

**Issue by mogeid** [pytorch/fairseq#3631]
#### Output changes between calls

I pre-trained a Roberta model on my own data.
After loading this model, its outputs seem to change from one call to the next on the same input.

#### Code

```python
feat1 = roberta.extract_features(inp[0])
feat2 = roberta.extract_features(inp[0])
print(feat1.shape == feat2.shape)
# > True
print(torch.all(inp[0] == inp[0]))
# > True
print(torch.any(feat1 == feat2))
# > tensor(False)
```
The difference is not insignificant:
```
torch.mean(torch.abs((feat1 - feat2)))
# > tensor(0.3770, grad_fn=<MeanBackward0>)
```

#### What have you tried?
I initially thought this was [an issue with padding](https://github.com/pytorch/fairseq/issues/3628#issue-924189960) changing the output of the encoder.
It seems however to be some kind of change between calls.
Is there a resetting between calls that I should be doing?
AFAIK, there's no state preserved between sequences and there's no randomness involved, so I don't understand why the output is changing.

#### What's your environment?

 - fairseq 1.0.0
 - PyTorch Version 1.7
 - OS: Linux

**Comment by mogeid** [comment]
forgot to call eval()
